src/relion/pipeline/__init__.py

```diff
@@ -697,7 +699,10 @@ def classification(
                         class_scores[class_scores >= 0], fraction_of_classes_to_remove
                     )
 
-                if not self.job_paths_batch.get(class2d_type):
+                if (
+                    not self.job_paths_batch.get(class2d_type)
+                    and fraction_of_classes_to_remove
+                ):
```

Comment:
I think this eliminates the first round of 2D classification entirely. Without class selection we only need to skip lines 721 to 732, and the current line 708 probably should move to there as well. (i.e. only an if on occurrences of "relion.select.class2D")